Applications/spire/source/time_and_sales/time_and_sales_properties_dialog.cpp

```diff
@@ -330,9 +332,46 @@ time_and_sales_properties_dialog::time_and_sales_properties_dialog(
   set_properties(properties);
 }
 
-const time_and_sales_properties&
+time_and_sales_properties
     time_and_sales_properties_dialog::get_properties() const {
-  return m_properties;
+  time_and_sales_properties properties;
+  auto unknown_item = m_band_list->item(0);
+  properties.set_band_color(price_range::UNKNOWN,
+    unknown_item->backgroundColor());
+  properties.set_text_color(price_range::UNKNOWN,
+    unknown_item->textColor());
+  auto above_ask_item = m_band_list->item(1);
+  properties.set_band_color(price_range::ABOVE_ASK,
+    above_ask_item->backgroundColor());
+  properties.set_text_color(price_range::ABOVE_ASK,
+    above_ask_item->textColor());
+  auto at_ask_item = m_band_list->item(2);
+  properties.set_band_color(price_range::AT_ASK,
+    at_ask_item->backgroundColor());
+  properties.set_text_color(price_range::AT_ASK,
+    at_ask_item->textColor());
+  auto inside_item = m_band_list->item(3);
+  properties.set_band_color(price_range::INSIDE,
+    inside_item->backgroundColor());
+  properties.set_text_color(price_range::INSIDE,
+    inside_item->textColor());
+  auto at_bid_item = m_band_list->item(4);
+  properties.set_band_color(price_range::AT_BID,
+    at_bid_item->backgroundColor());
+  properties.set_text_color(price_range::AT_BID,
+    at_bid_item->textColor());
+  auto below_bid_item = m_band_list->item(5);
+  properties.set_band_color(price_range::BELOW_BID,
+    below_bid_item->backgroundColor());
+  properties.set_text_color(price_range::BELOW_BID,
+    below_bid_item->textColor());
+  if(m_show_grid_check_box->isChecked()) {
```

Comment:
With boolean values, prefer to test/assign directly, ie:

  properties.m_show_grid = m_show_grid_check_box->isChecked();